src/maintagent/agent/process.c

```diff
@@ -24,81 +24,102 @@
 #include "maintagent.h"
 
 /**********  Globals  *************/
-extern PGconn    *pgConn;        ///< database connection
+PGconn* pgConn = NULL;        // the connection to Database
 
+/**
+ * \brief simple wrapper which includes PQexec and fo_checkPQcommand
+ * \param desc description for the SQL command, else NULL
+ * \param SQL  SQL command executed
+ * \param file source file name
+ * \param line source line number
+ * \return PQexec query result
+ */
+PGresult * PQexecCheck(int exitNumber, char *SQL, char *file, const int line)
+{
+  PGresult *result;
+  result = PQexec(pgConn, SQL);
+  if (fo_checkPQcommand(pgConn, result, SQL, file, line)) {
+    exitNow(exitNumber);
```

Comment:
Wouldn't it be better to clear the result before calling `exitNow()`? This can lead to memory leak.

Reply:
it will only enter the if statement it the rows are empty.